piff/gsobject_model.py

```diff
@@ -0,0 +1,389 @@
+from .util import hsm
+
+
+class GSObjectModel(Model):
+    """ Model that takes a fiducial GalSim.GSObject and dilates, shifts, and shears it to get a
+    good match to stars.
+
+    :param gsobj:    GSObject to use as fiducial profile.
+    :param fastfit:  Use HSM moments for fitting.  Approximate, but fast.  [default: False]
+    :param force_model_center: If True, PSF model centroid is fixed at origin and
+                        PSF fitting will marginalize over stellar position.  If False, stellar
+                        position is fixed at input value and the fitted PSF may be off-center.
+                        [default: True]
+    :param include_pixel:   Include integration over pixel?  [default: True]
+    :param logger:   A logger object for logging debug info. [default: None]
+    """
+    def __init__(self, gsobj, fastfit=False, force_model_center=True, include_pixel=True,
+                 logger=None):
+        import galsim
+        if isinstance(gsobj, basestring):
+            gsobj = eval(gsobj)
+
+        self.kwargs = {'gsobj':repr(gsobj),
+                       'fastfit':fastfit,
+                       'force_model_center':force_model_center,
+                       'include_pixel':include_pixel}
+
+        self.gsobj = gsobj
+        self._fastfit = fastfit
+        self._force_model_center = force_model_center
+        self._method = 'auto' if include_pixel else 'no_pixel'
+        # Params are [u, v], size, g1, g2
+        if self._force_model_center:
+            self._nparams = 3
+            params = [1.0, 0.0, 0.0]
+        else:
+            self._nparams = 5
+            params = [0.0, 0.0, 1.0, 0.0, 0.0]
+
+        # Calibrate gsobj by measuring it with HSM.  This way we can use differences in HSM moments
+        # to get a reasonable starting guess for stars.
+        prof = self.getProfile(params)
+        img = prof.drawImage(method=self._method)
+        sd = StarData(img, img.trueCenter())
+        fiducial_star = Star(sd, None)
+        flux, cenu, cenv, size, g1, g2, flag = hsm(fiducial_star)
+        if flag != 0:
+            raise RuntimeError("Error calculating fiducial moments for this gsobject.")
+        self._fiducial_size = size
+        self._fiducial_shape = galsim.Shear(g1=g1, g2=g2)
+        self._fiducial_flux = flux
+        self._fiducial_centroid = galsim.PositionD(cenu, cenv)
+
+    def moment_fit(self, star, logger=None):
+        """Estimate transformations needed to bring self.gsobj towards given star."""
+        import galsim
+        flux, cenu, cenv, size, g1, g2, flag = hsm(star)
+        centroid = galsim.PositionD(cenu, cenv)
+        shape = galsim.Shear(g1=g1, g2=g2)
+
+        if flag:
+            raise RuntimeError("Failed measuring HSM moments of star.")
+
+        dflux = flux/self._fiducial_flux
+        dcentroid = centroid - self._fiducial_centroid
+        dsize = size/self._fiducial_size
+        dshape = shape - self._fiducial_shape
```

Comment:
For Gaussian with sigma=1.0, the default drawImage scale is ~0.85.  Running HSM on the resulting image gives the flux, size, g1, and g2 params you list above.

If I force the scale to 0.2, then the HSM flux and size come out exactly 1.0.  The g1 and g2 are still a bit off from zero, but I'm sure we don't care about 10^-17 (although it is a bit weird, since the image is perfectly symmetric).

If I force the scale to 0.2 inside `GSObjectModel.__init__`, then all the unit tests pass with their old tolerances.  So I think the different wcses is the issue.  Only question then is how to address it.  Ironically, my first instinct would have been to use the `wmult` keyword just removed from GalSim.  In lieu of that, maybe bumping some of the accuracy parameters in gsparams would work instead?